content/_partials/video.html.haml

```diff
@@ -0,0 +1,21 @@
+- video_id = rand(1000)
+
+:css
+  .video_#{video_id} {
+    position: relative;
+    padding-bottom: 56.25%; /* 16:9 */
+    height: 0;
+  }
+  .video_#{video_id} iframe {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+  }
+  #ProjectVideo_#{video_id} {
+    background: #{video_background};
+  }
+
+.video.slide#ProjectVideo{:id => video_id, :class => "video_#{video_id}"}
```

Comment:
i'm pretty new to haml, but how does this translate to #ProjectVideo_1234? wouldn't it be 
`iframe id=1234 class="video_1234 video slide"`

I don't think an id is required anyways, since video_background is defined, and your not really referencing the id anywhere.

Reply:
The resulting HTML from that HAML fragment looks like this:

```
<!-- starting partial video.html.haml -->
--
<!-- starting partial video.html.haml -->
<style>
.video_990 {
position: relative;
padding-bottom: 56.25%; /* 16:9 */
height: 0;
}
.video_990 iframe {
position: absolute;
top: 0;
left: 0;
width: 100%;
height: 100%;
}
#ProjectVideo_990 {
background: ;
}
</style>
<div class="slide video video_990" id="ProjectVideo_990">
<iframe src="https://www.youtube.com/embed/_MXtbjwsz3A"></iframe>
</div>
 
<!-- ending partial video.html.haml -->
```

@halkeye based on what you know about CSS and JavaScript, does that HTML make sense?

Reply:
@halkeye looking more closely at the generated HTML, I have removed the reference to "slide" because there are no slides in this section.  That's a copy and paste error from the projectjumbotron page that I used as inspiration

Reply:
Generated HTML now looks like this:

```
<!-- starting partial video.html.haml -->
<style>
  .video_15 {
    position: relative;
    padding-bottom: 56.25%; /* 16:9 */
    height: 0;
  }
  .video_15 iframe {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
  }
</style>
<div class="video video_15" id="ProjectVideo">
<iframe allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" frameborder="0" height="621" src="https://www.youtube.com/embed/_MXtbjwsz3A" width="1261"></iframe>
</div>

<!-- ending partial video.html.haml -->
```

Reply:
So I would remove the Id. Ids have to be unique on the page, and unless you need to specifically target it, probably not worth specifying (I'll make a suggestion when I'm not mobile)

Otherwise it looks really good